vcr/verifier/verifier.go

```diff
@@ -136,10 +151,36 @@ func (v *verifier) Validate(credentialToVerify vc.VerifiableCredential, at *time
 		return fmt.Errorf("unable to resolve valid signing key at given time: %w", err)
 	}
 
-	// Try first with the correct LDProof implementation
 	return ldProof.Verify(signedDocument.DocumentWithoutProof(), signature.JSONWebSignature2020{ContextLoader: v.jsonldManager.DocumentLoader()}, pk)
 }
 
+func (v *verifier) validateJWTCredential(credential vc.VerifiableCredential, at *time.Time) error {
+	var keyID string
+	_, err := crypto.ParseJWT(credential.Raw(), func(kid string) (crypt.PublicKey, error) {
+		keyID = kid
+		return v.resolveSigningKey(kid, credential.Issuer.String(), at)
+	})
+	if err != nil {
+		return fmt.Errorf("unable to validate JWT credential: %w", err)
+	}
+	if keyID != "" && strings.Split(keyID, "#")[0] != credential.Issuer.String() {
+		return errVerificationMethodNotOfIssuer
+	}
+	return nil
+}
+
+func (v *verifier) resolveSigningKey(kid string, issuer string, at *time.Time) (crypt.PublicKey, error) {
+	// Compatibility: VC data model v1 puts key discovery out of scope and does not require the `kid` header.
+	// When `kid` isn't present use the JWT issuer as `kid`, then it is at least compatible with DID methods that contain a single verification method (did:jwk).
+	if kid == "" {
+		kid = issuer
+	}
+	if strings.HasPrefix(kid, "did:jwk:") && !strings.Contains(kid, "#") {
+		kid += "#0"
+	}
+	return v.keyResolver.ResolveKeyByID(kid, at, resolver.NutsSigningKeyType)
+}
```

Comment:
Is this `#0` according to did:jwk spec?

Why does it use a resolver.NutsSigningKeyType when it can also be a did:jwk?

Reply:
> Is this #0 according to did:jwk spec?

Yes.

> Why does it use a resolver.NutsSigningKeyType when it can also be a did:jwk?
maps to `assertionMethod`, which is now generally recognized as being the key for issuing VCs (but when we introduced the key, it wasn't). We can probably rename it (but out of scope).